src/lib/y2storage/disk_analyzer.rb

```diff
@@ -207,10 +214,34 @@ def find_linux_systems(disk)
       filesystems = linux_partitions(disk).map(&:filesystem)
       filesystems << disk.filesystem
       filesystems.compact!
-      return [] if filesystems.empty?
       filesystems.map { |f| release_name(f) }.compact
     end
 
+    # Finds fstab files in all filesystems
+    #
+    # @see #suitable_root_filesystems
+    #
+    # @return [Array<Fstab>]
```

Comment:
I would add a `TODO` comment somewhere around this functionality about (the lack of) caching. As we discussed offline, it's a pity that we are always re-mounting some partitions that were likely mounted already at a previous point of the installation by `#installed_systems`